students/hirot_directory/session10/oo_mailroom_hiro/test_mailroom.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+
+'''
+Hiro
+'''
+
+
+import os
+import pytest
+import mailroom
+
+
+# creates a sample database for the tests to use
+# sample_db = mailroom.DonorDB(mailroom.get_sample_data())
+
+# def test_empty_db():
+#     """
+#     tests that you can initilize an empty DB
+#     """
+#     db = mailroom.DonorDB()
+
+#     assert len(db.donors) == 0
+
+#     # donor_list = db.list_donors()
+#     # print(donor_list)
+#     # # no donors
+#     # assert donor_list.strip() == "Donor list:"
+
+
+def test_new_empty_donor():
+    """
+    creates an new donor with no donations
+    """
+    donor = mailroom.Donor("Fred Flintstone")
+
+    assert donor.name == "Fred Flintstone"
+    assert donor.last_donation is None
+
+
+def test_last_donation():
+    """
+    creates an new donor with no donations
+    """
+    donor = mailroom.Donor("Fred Flintstone", [100, 200, 300])
+
+    assert donor.last_donation == 300
+
+
+def test_add_donation():
+    # fixme: there should be a better way to get an arbitrary donor
+    donor = mailroom.Donor("Fred Flintstone", [100, 200, 300])
+
+    donor.make_donation(3000)
+
+    assert donor.last_donation == 3000
+
+
+def test_add_donor():
+    name = "Fred Flintstone  "
+
+    db = mailroom.DonorDB()
+
+    donor = db.add_donor(name)
+    donor.make_donation(300)
+    assert donor.name == "Fred Flintstone"
+    assert donor.last_donation == 300
+    # assert db.find_donor(name) == donor
+
+
+def test_find_donor():
+    """ checks a donor that is there, but with odd case and spaces"""
+    donor = mailroom.find_donor("jefF beZos ")
```

Comment:
does this work? I did a nifty normalization so all capitalization doesn't matter when searching for a donor. But you may want to start with having to ask for the exact name.

and when this test passes, make sure it's using the DonorDB find_donor() method!